### ISSUE TITLE
Boxy graphics when viewing with software renderer, even at native resolution

### ISSUE THREAD

**Issue by chaitanyathengdi** [OpenRCT2/OpenRCT2#12306]
**OS:** Any(tested with Windows 7 and 10)
**Version:** 0.2.6
**Commit/Build:** `6c3c857`

Graphics appear "boxy" when using the software renderer(also has higher fps).
Normal graphics when using software with hardware display.
This issue is not showable using the in-game screenshot feature, which shows normal rendering.

- [ ] Reproducible in RCT2 (vanilla)? *option is not there in vanilla*
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Set the display type to Software in the Options window.
2. Set the Scale Factor to anything other(lesser or higher) than 1.00

**Screenshots:**(open in new tab to see the problem)
**Using in-game screenshot feature**
![image](https://user-images.githubusercontent.com/14993220/87858645-0adad600-c94d-11ea-940f-2fab969d511d.png)
![image](https://user-images.githubusercontent.com/14993220/87858654-14fcd480-c94d-11ea-866f-58803cc58a2d.png)

**Using Print Screen**
![image](https://user-images.githubusercontent.com/14993220/87858859-93a64180-c94e-11ea-9c28-1ba726ec9997.png)
![image](https://user-images.githubusercontent.com/14993220/87858886-be909580-c94e-11ea-9ddd-c7062ee74924.png)

**Comment by chaitanyathengdi** [comment]
Does the software renderer miss a step or something? Like anti-aliasing?

**Comment by IntelOrca** [comment]
Yes, I think you need hardware disable for linear filtering at non-integral scale factors. V-sync is also not available for software rendering because it only uses OS APIs.

Software (hardware display) is available to solve both of these issues as well as others.

**Comment by chaitanyathengdi** [comment]
I've noticed this is an issue when the scale factor is small (smaller than ~3) and is a fraction(i.e. not a multiple of 1.00) in general. If the factor is too high, it matches anyway.

**Comment by chaitanyathengdi** [comment]
Couldn't we do anything about it? I'd really like to use software because hardware-display software has that issue of black screen when switching windows. I could get away with OpenGL on my Win7 desktop but not on my Win10 laptop.

**Comment by IntelOrca** [comment]
@chaitanyathengdi does that issue happen if you use borderless fullscreen?

**Comment by frutiemax** [comment]
I also have this issue with software rendering when scaling is higher than 1.

**Comment by janisozaur** [comment]
The reason it's solved in software+hw is we render to an off-screen buffer using `ceil(scale)` and scale it back down to `scale`. Doing the same purely in software would incur too high cost.

What's your current FPS at desired scale?

**Comment by chaitanyathengdi** [comment]
In excess of 150, even 250 in some cases, if park is mostly empty(e.g. beginning of Forest Frontiers).
In case of highly populated park, like below: ~90
![image](https://user-images.githubusercontent.com/14993220/87861867-1ab3e380-c968-11ea-88d6-696cbf6fe554.png)
Goes to around 60 if zoomed out all the way.

**Comment by chaitanyathengdi** [comment]
> does that issue happen if you use borderless fullscreen?
@IntelOrca Yes, it's there even then.

**Comment by IntelOrca** [comment]
@chaitanyathengdi does it happen if you are running the game in a window?

**Comment by janisozaur** [comment]
@chaitanyathengdi can you post some recording of the black screen issue?

**Comment by chaitanyathengdi** [comment]
>does it happen if you are running the game in a window?

@IntelOrca Yes it does. Here:
![image](https://user-images.githubusercontent.com/14993220/87875625-29de7400-c9f0-11ea-9829-9f1bee38956e.png)

**Comment by IntelOrca** [comment]
@chaitanyathengdi I am talking about the black screen issue.

**Comment by chaitanyathengdi** [comment]
@IntelOrca Oh. No, only in fullscreen. Windowed or borderless windowed are fine.

**Comment by chaitanyathengdi** [comment]
@janisozaur Here's a video:
https://imgur.com/clUQqoC
As you can see in the end, it just goes black.

**Comment by IntelOrca** [comment]
@chaitanyathengdi If borderless fullscreen is fine then that's what you should continue using alongside hardware display.

**Comment by Derpercoaster** [comment]
RCT was built around fixed resolutions (the resolutions of the monitor), not software zooming. The game is being forced to resize graphics that weren't meant to be resized, which causes the artifacts. At least, I think this is what is happening.
